File: 07-stacks-queues/08_largest_rect_histogram.c
#include "../test_runner.h"

int largest_rectangle_area(int* heights, int n) {
    return -1;
}

int main() {
    TEST_INT("basic",  largest_rectangle_area(ARR(2,1,5,6,2,3), 6), 10);
    TEST_INT("two",    largest_rectangle_area(ARR(2,4), 2),          4);
    TEST_INT("single", largest_rectangle_area(ARR(1), 1),            1);
    TEST_SUMMARY();
}

File: 07-stacks-queues/08_largest_rect_histogram.py
import sys, os; sys.path.append(os.path.join(os.path.dirname(os.path.abspath(__file__)), '..'))
from test_runner import run_tests


# Largest Rectangle in Histogram (LC #84) -- Hard
# Largest rectangle that can be formed in histogram.
#
#   Key: Monotonic increasing stack. For each bar, find how far it extends left and right.

def largest_rectangle_area(heights):
    pass


run_tests(largest_rectangle_area, [
    (([2,1,5,6,2,3],), 10),
    (([2,4],),           4),
    (([1],),             1),
])
